android_jni/avifandroidjni/src/main/jni/libavif_jni.cc

```diff
@@ -175,24 +183,39 @@ FUNC(jboolean, isAvifImage, jobject encoded, int length) {
   return avifPeekCompatibleFileType(&avif);
 }
 
+#define CHECK_EXCEPTION(ret) \
+  if (JniExceptionCheck(env)) return ret
+
+#define FIND_CLASS(var, class_name, ret)         \
+  const jclass var = env->FindClass(class_name); \
+  CHECK_EXCEPTION(ret);                          \
+  if (var == nullptr) return ret
+
+#define GET_FIELD_ID(var, class_name, field_name, signature, ret)          \
+  const jfieldID var = env->GetFieldID(class_name, field_name, signature); \
+  CHECK_EXCEPTION(ret);                                                    \
+  if (var == nullptr) return ret
```

Comment:
Should these three macros be put inside a `do {} while (false)` block so that they behave like a statement?

Reply:
We cannot do that here since these macros declare a local variable which has to be visible in the calling scope.